action-server/alembic/versions/00002_test_profiles.py

```diff
@@ -0,0 +1,278 @@
+        symptoms="moderate", has_fever=True, has_cough=True, has_diff_breathing=True
+    )
+)
+
+TEST_PROFILES = [TEST_PROFILE_1, TEST_PROFILE_2, TEST_PROFILE_3, TEST_PROFILE_4]
+
+
+def upgrade():
+    reminder_table = _create_reminder_table_construct()
+    assessment_table = _create_assessment_table_construct()
+
+    _create_test_profiles(reminder_table, assessment_table, TEST_PROFILES)
+
+
+def downgrade():
+    _delete_test_profiles()
+
+
+def _create_reminder_table_construct():
+    log.info("Creating reminder table construct.")
+
+    return table(
+        REMINDER_TABLE,
+        Column("id", Integer, primary_key=True),
+        Column(
+            "created_at",
+            DateTime(timezone=True),
+            server_default=func.current_timestamp(),
+            nullable=False,
+        ),
+        Column("last_reminded_at", DateTime(timezone=True),),
+        Column("timezone", String(), CheckConstraint("is_timezone(timezone)"),),
+        Column("is_canceled", Boolean(), nullable=False, server_default="false"),
+        Column("attributes", JSONB, nullable=False),
+    )
+
+
+def _create_assessment_table_construct():
+    log.info("Creating assessment table construct.")
+
+    return table(
+        ASSESSMENT_TABLE,
+        Column("id", Integer, primary_key=True),
+        Column(
+            "reminder_id",
+            Integer,
+            ForeignKey("reminder.id", onupdate="CASCADE", ondelete="CASCADE"),
+        ),
+        Column(
+            "completed_at",
+            DateTime(timezone=True),
+            server_default=func.current_timestamp(),
+            nullable=False,
+        ),
+        Column("attributes", JSONB, nullable=False),
+    )
+
+
+def _create_test_profiles(reminder_table, assessment_table, profiles):
+    log.info("Adding test profiles to reminder table.")
+
+    reminders = [test_profile.create_reminder() for test_profile in profiles]
+
+    op.bulk_insert(reminder_table, reminders)
+
+    log.info("Fetching test profiles reminder ids.")
+
+    connection = op.get_bind()
+    reminders = connection.execute(
+        f"select id, attributes->>'phone_number' from {REMINDER_TABLE} where attributes->>'test_profile' = '{TEST_PROFILE_REVISION}';"
+    )
+
+    log.info("Adding test profiles to assessment table.")
+
+    assessments = []
+
+    for reminder in reminders:
+        reminder_id = reminder[0]
+        phone_number = reminder[1]
+
+        hashed_reminder_id = encode_reminder_id(reminder.id)
+
+        log.info(
+            f"reminder_id={reminder_id}, hashed reminder_id={hashed_reminder_id}, phone_number={phone_number}"
+        )
+
+        profile = next(
+            (
+                profile
+                for profile in profiles
+                if profile.reminder_attributes["phone_number"] == phone_number
+            ),
+            None,
+        )
+
+        if profile is None:
+            raise Exception(f"Could not find profile for phone number {phone_number}.")
```

Comment:
What happens if this is raised?

Reply:
migration will fail I guess... but this shouldn't really happen unless there's an error in the code itself (or if we have bigger problems, like connection issues). If that ever happens, downgrading will remove inconsistancies incorporated by this (failing) migration.